### ISSUE TITLE
Log a warning when calling grid.setItems() with a large list

### ISSUE THREAD

**Issue by marcushellberg** [vaadin/flow-components#4704]
### Describe your motivation

Using large in-memory lists to back Grids is a common cause for excessive memory usage.

### Describe the solution you'd like

Consider logging a warning in the console if `setItems()` is called with a list of more than N items. Something like "Calling setItems(list) with a large in-memory list can cause performance issues, consider using setItems(q->{})"

### Describe alternatives you've considered

Hoping that users know what they're doing.

**Comment by knoobie** [comment]
I would argue that this highly depends on the backend. I've seen Grid's where it was literally light years faster to instantly load all data from the database instead of using lazy loading because of the additional costs for round trips over the wire / external services and slow and really expensive Queries.

If you really wanna add this warning, please do it on INFO or allow disabling it.

**Comment by sissbruecker** [comment]
Yeah, this is a trade-off between speed and memory utilization that a developer needs to make. I get the concern, but I'm not sure if a warning is an appropriate solution here.

**Comment by web-padawan** [comment]
To me it sounds like a DX related issue that might require improving documentation with a corresponding note.

**Comment by TatuLund** [comment]
I tend to agree with @knoobie. I have seen that this is highly application specific question. Also noting here, that doing lazy loading from server memory (which is done always by the Grid) is usually faster than doing the query on the database level. In many real life applications the database is the most stressed component. However from scalability point of view it is better to make session footprint as small as possible. So if that is the optimization goal, then doing lazy loading from database is the thing to do.
